packages/cli/commands/theme/marketplace-validate.js

```diff
@@ -100,20 +100,54 @@ exports.handler = async options => {
   }
 
   const displayResults = checks => {
+    const displayFileInfo = (file, line) => {
+      file &&
+        logger.log(
+          i18n(`${i18nKey}.results.warnings.file`, {
+            file,
+          })
+        );
+      line &&
+        logger.log(
+          i18n(`${i18nKey}.results.warnings.lineNumber`, {
+            line,
+          })
+        );
+    };
+
     if (checks) {
       const { status, results } = checks;
 
       if (status === 'FAIL') {
         const failedValidations = results.filter(
           test => test.status === 'FAIL'
         );
+        const warningValidations = results.filter(
+          test => test.status === 'WARN'
+        );
+
         failedValidations.forEach(val => {
           logger.error(`${val.message}`);
+          displayFileInfo(val.file, val.line);
+        });
+
+        warningValidations.forEach(val => {
+          logger.warn(`${val.message}`);
+          displayFileInfo(val.file, val.line);
         });
       }
 
       if (status === 'PASS') {
         logger.success(i18n(`${i18nKey}.results.noErrors`));
+
+        const warningValidations = results.filter(
+          test => test.status === 'WARN'
+        );
+
+        warningValidations.forEach(val => {
```

Comment:
You could cut down on lines of code a little here by using an if check for `val.status === 'warn'` in the `forEach`